### PR TITLE
Delete zero-copy locks

### PR DESCRIPTION
**Pull request by kirillgarbar**
## Summary by Sourcery

Add a new chadmin zookeeper subcommand for cleaning up zero-copy replication locks in ZooKeeper, integrate it into host removal routines, and expand tests to cover these scenarios.

New Features:
- Introduce cleanup-zero-copy-locks CLI command to remove zero-copy replication locks with table, part and replica filters and dry-run support
- Automatically trigger zero-copy lock cleanup when removing hosts via chadmin

Enhancements:
- Add batched zero-copy lock deletion logic and helper to retrieve zero-copy ZooKeeper path from ClickHouse settings

Tests:
- Add Cucumber scenarios testing various zero-copy lock cleanup flows and update step definitions to correct singular grammar in zookeeper list assertions

### CODE REVIEW

File: ch_tools/chadmin/internal/zookeeper_clean.py [hunk 1]
```diff
@@ -538,3 +542,98 @@ def mark_finished_ddl_query(zk):
...
+        paths = find_paths(zk, zero_copy_path, [template])
+        if not paths:
+            return
+        table_path = os.path.dirname(paths[0])
+        # Checking if we can just delete the table's directory instead
+        if len(get_children(zk, table_path)) == 1:
+            paths = [table_path]
+    else:
+        template = re.escape(rf"{zero_copy_path}/{table_uuid}")
+        paths = find_paths(zk, zero_copy_path, [template])
```
[the remark this answers is not in the document]
Comment (kirillgarbar) [reply ch_tools/chadmin/internal/zookeeper_clean.py:562]: It is true, this means `/clickhouse/zero_copy/zero_copy_s3` will not be deleted. I am not sure if we want to delete it.

**Review by aalexfvk — APPROVED** [review]
